### PR TITLE
Added KMS support to s3Upload

### PR DESCRIPTION
**Pull request by meve**
* **Please check if the PR fulfills these requirements**
- [x] The commit message describes your change
- [x] Tests for the changes have been added if possible (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
I tried adding support for KMS to the s3Upload method.

* **What is the current behavior?** (You can also link to an open issue here)
The current behavior is that this is not implemented.

* **What is the new behavior (if this is a feature change)?**
The new behavior is that you can add a parameter to s3Upload to specify either a KMS alias or KMS id.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)
Not that I know of.

* **Other information**:
This is my first PR. Any feedback is welcome. Thank you in advance.

### CODE REVIEW

File: src/main/java/de/taimos/pipeline/aws/S3UploadStep.java [hunk 1]
```diff
@@ -329,12 +341,22 @@ public Void invoke(File localFile, VirtualChannel channel) throws IOException, I
 					if (this.contentType != null && !this.contentType.isEmpty()) {
 						metas.setContentType(contentType);
 					}
-					PutObjectRequest request = new PutObjectRequest(this.bucket, this.path, localFile).withMetadata(metas);
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/S3UploadStep.java]: You can leave this, and just add the KMS config when the id is set.

File: src/main/java/de/taimos/pipeline/aws/S3UploadStep.java [hunk 2]
```diff
@@ -443,6 +475,15 @@ public void provideObjectMetadata(File file, ObjectMetadata meta) {
 						if (RemoteListUploader.this.contentType != null && !RemoteListUploader.this.contentType.isEmpty()) {
 							meta.setContentType(RemoteListUploader.this.contentType);
 						}
+						if (RemoteListUploader.this.kmsId != null && !RemoteListUploader.this.contentType.isEmpty()) {
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/S3UploadStep.java]: should be kmsId.isEmpty()

**Comment by meve** [comment]
Thank you for you feedback @hoegertn. I have modified the code according to your comments.

**Comment by hoegertn** [comment]
Thanks, I will merge it and rework the part where the metas, acl and kms are handled as it is too complicated and in some cases wrong imho. But this is not your fault and was broken before.

**Comment by meve** [comment]
@hoegertn Wait. I think I have covered that as well now.

**Comment by meve** [comment]
@hoegertn Okay, I think it's good now.

**Comment by hoegertn** [comment]
Perfect, that is exactly what I had in mind.

**Pull request merged**